### PR TITLE
`CellIterator` broadcast safety doc warning

### LINKED ISSUE #267: CellIterator not broadcast safe
I haven't looked into the details of that issue (Opened it as a reminder) and tbh I don't know much about the inner workings of broadcast and materialize. As the current id is mutated in the `CellIterator` the first example only picks up the last id. Any ideas if there's a possible fix for that?

### Wrong
```
julia> cellid.(CellIterator(dh))
22754-element Array{Int64,1}:
 22754
 22754
 22754
 22754
 22754
 22754
     ⋮
 22754
 22754
 22754
 22754
 22754
```

### Correct
```
julia> map(CellIterator(dh)) do cell
           cellid(cell)
       end
22754-element Array{Int64,1}:
     1
     2
     3
     4
     5
     6
     ⋮
 22750
 22751
 22752
 22753
 22754
```

### PR DESCRIPTION
**Pull request by AbdAlazezAhmed**
Adds warning to docstring

### CODE REVIEW

**Comment by fredrikekre** [comment]
I don't think we should do this -- it is too easily circumvented (`[c for c in CellIterator(dh)]` for example), and is just not how `CellIterator` is supposed to be used. Perhaps we can just add a note in the docstring that the iterator is stateful?

**Comment by AbdAlazezAhmed** [comment]
Ah, I get it, I thought it could have some use case. Maybe we can throw an error then?
```julia
julia> cellid.(CellIterator(dh))
ERROR: CellIterator is stateful and should not be used in broadcasting or collected
```

**Comment by fredrikekre** [comment]
I think I prefer just the note in the documentation, it isn't feasible to catch every misuse like this anyway.

File: src/iterators.jl [hunk 1]
```diff
@@ -128,6 +128,10 @@ for idx in cellset
     # ...
 end
 ```
+!!! warning
+
+    `CellIterator` is stateful and should not be used in broadcasting nor collected
+
```
Comment (fredrikekre) [inline src/iterators.jl:134]:
```suggestion
!!! warning
    `CellIterator` is stateful and should not be used for things other than `for`-looping
    (e.g. broadcasting over, or collecting the iterator may yield unexpected results).
```

**Pull request merged**